design/m1_push_to_file_design.md

```diff
@@ -0,0 +1,1302 @@
+
+- [ ] All items in the
+  [Development Tasks: Minimally-Featured GA Release](#development-tasks-minimally-featured-ga-release)
+  section for the will be implemented with unit tests.
+- [ ] Pet store demo app supports getting its database configuration via input file.
+- [ ] There is a happy path e2e test in the Secrets Provider test cases to validate
+  using the K8s Secrets init container with annotation-based configuration
+- [ ] There is a happy path e2e test in the Secrets Provider test cases to
+  validate using the K8s Secrets Job with annotation-based configuration
+- [ ] There are e2e tests in supported OpenShift versions and GKE with Secrets Provider init container in "file" mode running with:
+    - Conjur Open Source
+    - Conjur Enterprise with follower in Kubernetes
+    - Conjur Enterprise with follower outside Kubernetes
+- [ ] Init container and Job flows have both been tested
+- [ ] There is documentation collateral for the TWs
+- [ ] Dap-Wiki has been updated with info on the new flows
+- [ ] There is basic troubleshooting information available
+- [ ] Any custom upgrade instructions are documented
+- [ ] Instructions for restarting pod on annotations changes have been documented
+- [ ] Documentation is clear about requirements for volume mapping and file permissions
+- [ ] (TBD) Performance of the SP with the "file" flow supported has been measured, and has not appreciably decreased from prior measurements (prior perf tests)
+- [ ] SP has been released with the new "file" functionality, including manual security scans
+
+### Full-Featured (Certified) Community Release
+
+- [ ] All items in the
+  [Deployment Tasks: Full-Featured (Certified) Community Release](#deployment-tasks-full-featured-certified-community-release)
+  section will be implemented with unit tests.
+- [ ] There is a happy path e2e test in the Secrets Provider test cases to validate
+  using annotation-based configuration without secrets aliases
+- [ ] There is a happy path e2e test in the Secrets Provider test cases to validate
+  using secrets path prefix
+- [ ] There is a happy path e2e test in the Secrets Provider test cases to validate
+  writing of JSON formatted file
+- [ ] There is a happy path e2e test in the Secrets Provider test cases to validate
+  using an output file path/name
+- [ ] There is a happy path e2e test in the Secrets Provider test cases to validate
+  writing of Bash formatted file
+- [ ] There is a happy path e2e test in the Secrets Provider test cases to validate
+  writing of dotenv formatted file
+- [ ] There is a happy path e2e test to validate Helm named templates for init
+  container def, volume mounts, and volumes
+- [ ] There is a blog post describing how to transition from an existing application to using the new Secrets Provider "push to file" flow
+- [ ] There is a recorded demo of the new SP "file" functionality
+- [ ] The new SP "file" functionality has been shared at a training session
+
```

Comment:
Future work: 
- [ ] Add ability to configure secret file permissions via `conjur.org/secret-file-perms.{secret-group}` annotation